### PR TITLE
meetinga

### LINKED ISSUE #60: login from meetinga

### PR DESCRIPTION
**Pull request by qunabu**
Closes #60

### CODE REVIEW

File: app/Http/Middleware/Meetinga.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    {
+
+        return array_map(
+            function ($group) use ($mainGroup) {
+                return Group::firstOrCreate([
+                    'name' => $group['name'],
+                    'registerable' => true,
+                    'parent_id' => $mainGroup->id
+                ]);
+            },
```
Comment (pa-cholek) [inline app/Http/Middleware/Meetinga.php:31]:
```suggestion
            fn ($group) => Group::firstOrCreate([
                    'name' => $group['name'],
                    'registerable' => true,
                    'parent_id' => $mainGroup->id
                ])
```

File: app/Http/Middleware/Meetinga.php [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        $userSettings = collect($data)->only(['city', 'postal_code', 'country', 'gender'])->toArray();
+
+        $mainGroup = Group::firstOrCreate([
+            'name' => 'Nauczyciele',
+            'registerable' => true
+        ]);
+
+        $groups = [$mainGroup];
+
+        $groups = array_merge($groups, $this->createOrFindGroups($mainGroup, $data['user_groups1']));
```
Comment (pa-cholek) [inline app/Http/Middleware/Meetinga.php:72]: Will data always have these three keys/values? (`user_groups1` etc)

**Review by pa-cholek — APPROVED** [review]

File: app/Http/Middleware/Meetinga.php (hunk 2 above)
Comment (qunabu) [reply app/Http/Middleware/Meetinga.php:72]: yes. it's hardcoded

File: app/Http/Middleware/Meetinga.php [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    private function createOrFindGroups(Model $mainGroup, array $groups): array
+    {
+
+        return array_map(
+            fn ($group) => Group::firstOrCreate([
+                    'name' => $group['name'],
+                    'registerable' => true,
+                    'parent_id' => $mainGroup->id
+                ])
```
Comment (pa-cholek) [inline app/Http/Middleware/Meetinga.php:29]:
```suggestion
                ]),
```

**Pull request merged**